extensions/wait/wait.go

```diff
@@ -0,0 +1,33 @@
+package wait
+
+import (
+	"github.com/rancher/shepherd/pkg/wait"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func ResourceCreation(watchInterface watch.Interface) error {
```

Comment:
nit; the name I would expect is `ResourceCreate` and `ResourceDelete` though I don't know if we have rules around this. Most other names follow this, though.

Reply:
Good point. I'll rename them